core/src/main/scala/org/apache/spark/ui/exec/ExecutorsTab.scala

```diff
@@ -68,6 +72,7 @@ class ExecutorsListener(storageStatusListener: StorageStatusListener) extends Sp
   override def onExecutorRemoved(
       executorRemoved: SparkListenerExecutorRemoved): Unit = synchronized {
     val eid = executorRemoved.executorId
+    executorIds = executorIds.filter(x => (x != eid))
```

Comment:
I can't make out how this relates to the problem you specified.  Do you mind backing up and explaining in the JIRA first? I would close this PR as it does not look like the right approach to anything like what you're describing